src/System.Net.Security/src/System/Net/SslStreamPal.Windows.cs

```diff
@@ -129,6 +128,31 @@ public static SafeFreeCredentials AcquireCredentialsHandle(X509Certificate certi
 
         public static SecurityStatusPal EncryptMessage(SafeDeleteContext securityContext, byte[] input, int offset, int size, int headerSize, int trailerSize, ref byte[] output, out int resultSize)
         {
+            // Ensure that there is sufficient space for the message output.
+            try
+            {
+                int bufferSizeNeeded = checked(size + headerSize + trailerSize);
+
+                if (output == null || output.Length < bufferSizeNeeded)
+                {
+                    output = new byte[bufferSizeNeeded];
+                }
+            }
+            catch (Exception e)
+            {
+                if (!ExceptionCheck.IsFatal(e))
+                {
+                    if (GlobalLog.IsEnabled)
+                    {
+                        GlobalLog.Assert("SslStreamPal.Windows: SecureChannel#" + LoggingHash.HashString(securityContext) + "::Encrypt", "Arguments out of range.");
+                    }
+
+                    Debug.Fail("SslStreamPal.Windows: SecureChannel#" + LoggingHash.HashString(securityContext) + "::Encrypt", "Arguments out of range.");
```

Comment:
Looks good!
GlobalLog was enabled only in CoreFX and its purpose is to debug framework issues rather than customer issues so I don't think we have an app-compat issue. (In the Desktop Framework GlobalLog is enabled in debug builds only.)